### PR TITLE
APM > .NET > Setting env vars for Windows Services

### PR DESCRIPTION
**Pull request by andrewardito**
### What does this PR do?
Add instructs to dotnet setup to ensure docs readibility

### Preview link
https://docs-staging.datadoghq.com/andrew.ardito/dotnetupdates/tracing/setup/dotnet-framework/?tab=environmentvariables#required-environment-variables

### CODE REVIEW

File: content/en/tracing/setup/dotnet-framework.md [hunk 1]
```diff
@@ -64,26 +64,47 @@ If your application runs in IIS, skip the rest of this section.
...
+- Value Name: Environment
+- Value Data:
+CORECLR_ENABLE_PROFILING=1
+CORECLR_PROFILER={846F5F1C-F9AE-4B07-969E-05C26BC060D8}
+
+This can be done either through the Registry Editor as in the image below, or through a PowerShell snippet:
+
+{{< img src="tracing/setup/dotnet/RegistryEditor.png" alt="Registry Editor"  >}}
+
+{{< code-block lang="dotnet" filename="PowerShell" >}}
```
Comment (lucaspimentel) [inline content/en/tracing/setup/dotnet-framework.md:89]: The correct `lang` is going to be either `ps` or `powershell`. I'm not sure which one, so let's try this.
```suggestion
{{< code-block lang="ps" filename="PowerShell" >}}
```

File: content/en/tracing/setup/dotnet-core.md [hunk 2]
```diff
@@ -126,12 +126,36 @@ If your application runs in IIS, skip the rest of this section.
...
+- Value Name: Environment
+- Value Data:
+CORECLR_ENABLE_PROFILING=1
+CORECLR_PROFILER={846F5F1C-F9AE-4B07-969E-05C26BC060D8}
+
+This can be done either through the Registry Editor as in the image below, or through a PowerShell snippet:
+
+{{< img src="tracing/setup/dotnet/RegistryEditor.png" alt="Registry Editor"  >}}
+
+{{< code-block lang="dotnet" filename="PowerShell" >}}
```
Comment (lucaspimentel) [inline content/en/tracing/setup/dotnet-core.md:150]: The correct `lang` is going to be either `ps` or `powershell`. I'm not sure which one, so let's try this.
```suggestion
{{< code-block lang="ps" filename="add-env-var.ps1" >}}
```

File: content/en/tracing/custom_instrumentation/dotnet.md [hunk 3]
```diff
@@ -17,9 +17,11 @@ further_reading:
       text: 'Explore your services, resources, and traces'
 ---
 <div class="alert alert-info">
-If you have not yet read the instructions for auto-instrumentation and setup, start with the <a href="https://docs.datadoghq.com/tracing/setup/dotnet/">.NET Setup Instructions</a>.
+If you have not yet read the instructions for auto-instrumentation and setup and are looking for those, you can find them here: <a href="https://docs.datadoghq.com/tracing/setup/dotnet/">.NET Setup Instructions</a>.
```
Comment (lucaspimentel) [inline content/en/tracing/custom_instrumentation/dotnet.md:20]: How about something like:

```suggestion
For instructions on how to setup the .NET Tracer and enable automatic instrumentation, see the [.NET Setup Instructions][???].
```

(note also I changed the link syntax, but it needs adding the link to the list below).

File: content/en/tracing/setup/dotnet-core.md [hunk 4]
```diff
@@ -126,12 +126,37 @@ If your application runs in IIS, skip the rest of this section.
 
 For Windows applications **not** running in IIS, set these two environment variables before starting your application to enable automatic instrumentation:
 
+**Note:** The .NET runtime tries to load a profiler into _any_ .NET process that is started with these environment variables are set. You should limit instrumentation only to the applications that need to be traced. **We do not recommend setting these environment variables globally as this causes _all_ .NET processes on the host to load the profiler.**
```
Comment (lucaspimentel) [inline content/en/tracing/setup/dotnet-core.md:129]:
```suggestion
**Note:** The .NET runtime tries to load a profiler into _any_ .NET process that is started with these environment variables set. You should limit instrumentation only to the applications that need to be traced. **We do not recommend setting these environment variables globally as this causes _all_ .NET processes on the host to load the profiler.**
```

File: content/en/tracing/setup/dotnet-core.md [hunk 5]
```diff
@@ -126,12 +126,37 @@ If your application runs in IIS, skip the rest of this section.
...
+ CORECLR_PROFILER={846F5F1C-F9AE-4B07-969E-05C26BC060D8}
+
+
+This can be done either through the Registry Editor as in the image below, or through a PowerShell snippet:
+
+{{< img src="tracing/setup/dotnet/RegistryEditor.png" alt="Registry Editor"  >}}
+
+{{< code-block lang="powershell" filename="add-env-var.ps1" >}}
+[String[]] $v = @("COR_ENABLE_PROFILING=1", "COR_PROFILER={846F5F1C-F9AE-4B07-969E-05C26BC060D8}")
+
```
Comment (lucaspimentel) [inline content/en/tracing/setup/dotnet-core.md:151]: Let's remove the blank line.
```suggestion

```

File: content/en/tracing/setup/dotnet-framework.md [hunk 6]
```diff
@@ -64,26 +64,45 @@ If your application runs in IIS, skip the rest of this section.
 
 For Windows applications **not** running in IIS, set these two environment variables before starting your application to enable automatic instrumentation:
 
+
+**Note:** The .NET runtime tries to load a profiler into _any_ .NET process that is started with these environment variables are set. You should limit instrumentation only to the applications that need to be traced. **We do not recommend setting these environment variables globally as this causes _all_ .NET processes on the host to load the profiler.**
```
Comment (lucaspimentel) [inline content/en/tracing/setup/dotnet-framework.md:68]:
```suggestion
**Note:** The .NET runtime tries to load a profiler into _any_ .NET process that is started with these environment variables set. You should limit instrumentation only to the applications that need to be traced. **We do not recommend setting these environment variables globally as this causes _all_ .NET processes on the host to load the profiler.**
```

File: content/en/tracing/setup/dotnet-framework.md [hunk 7]
```diff
@@ -64,26 +64,45 @@ If your application runs in IIS, skip the rest of this section.
...
+CORECLR_PROFILER={846F5F1C-F9AE-4B07-969E-05C26BC060D8}
+```
+
+This can be done either through the Registry Editor as in the image below, or through a PowerShell snippet:
+
+{{< img src="tracing/setup/dotnet/RegistryEditor.png" alt="Registry Editor"  >}}
+
+{{< code-block lang="powershell" filename="add-env-var.ps1" >}}
+[String[]] $v = @("COR_ENABLE_PROFILING=1", "COR_PROFILER={846F5F1C-F9AE-4B07-969E-05C26BC060D8}")
+
```
Comment (lucaspimentel) [inline content/en/tracing/setup/dotnet-framework.md:89]: Let's remove the blank line.
```suggestion

```

File: content/en/tracing/setup/dotnet-framework.md [hunk 8]
```diff
@@ -64,26 +64,45 @@ If your application runs in IIS, skip the rest of this section.
...
 ```bat
 rem Set environment variables
-SET COR_ENABLE_PROFILING=1
-SET COR_PROFILER={846F5F1C-F9AE-4B07-969E-05C26BC060D8}
+SET CORECLR_ENABLE_PROFILING=1
+SET CORECLR_PROFILER={846F5F1C-F9AE-4B07-969E-05C26BC060D8}
 
 rem Start application
-example.exe
+dotnet.exe example.dll
```
Comment (lucaspimentel) [inline content/en/tracing/setup/dotnet-framework.md:103]: Revert these changes, please. The previous instructions were correct for .NET Framework.

File: content/en/tracing/setup/dotnet-framework.md [hunk 9]
```diff
@@ -64,26 +64,45 @@ If your application runs in IIS, skip the rest of this section.
...
 | `COR_ENABLE_PROFILING` | `1`                                      |
 | `COR_PROFILER`         | `{846F5F1C-F9AE-4B07-969E-05C26BC060D8}` |
 
-For example, to set the environment variables from a batch file before starting your application:
+#### Windows Services
+
+To automatically instrument a Windows Service, set the environment variables for that Service in the Windows Registry. Create a multi-string value called `Environment` in the `HKLM\System\CurrentControlSet\Services\<Service Name>` key. Then set the key's data to the values in the table:
+```text
+CORECLR_ENABLE_PROFILING=1
+CORECLR_PROFILER={846F5F1C-F9AE-4B07-969E-05C26BC060D8}
```
Comment (lucaspimentel) [inline content/en/tracing/setup/dotnet-framework.md:80]: The env var names are slightly different between .NET Framework and .NET Core.
```suggestion
COR_ENABLE_PROFILING=1
COR_PROFILER={846F5F1C-F9AE-4B07-969E-05C26BC060D8}
```

File: content/en/tracing/setup/dotnet-core.md [hunk 10]
```diff
@@ -126,12 +126,35 @@ If your application runs in IIS, skip the rest of this section.
...
+CORECLR_PROFILER={846F5F1C-F9AE-4B07-969E-05C26BC060D8}
+```
+
+
+This can be done either through the Registry Editor as in the image below, or through a PowerShell snippet:
+
+{{< img src="tracing/setup/dotnet/RegistryEditor.png" alt="Registry Editor"  >}}
+
+{{< code-block lang="powershell" filename="add-env-var.ps1" >}}
+[String[]] $v = @("COR_ENABLE_PROFILING=1", "COR_PROFILER={846F5F1C-F9AE-4B07-969E-05C26BC060D8}")
```
Comment (lucaspimentel) [inline content/en/tracing/setup/dotnet-core.md:150]: The env var names are slightly different between .NET Framework and .NET Core.
```suggestion
[String[]] $v = @("CORECLR_ENABLE_PROFILING=1", "CORECLR_PROFILER={846F5F1C-F9AE-4B07-969E-05C26BC060D8}")
```

File: content/en/tracing/setup/dotnet-core.md (hunk 4 above)
Comment (lucaspimentel) [reply content/en/tracing/setup/dotnet-core.md:129]: Unresolving since it's still there :)

File: content/en/tracing/setup/dotnet-core.md [hunk 11]
```diff
@@ -126,24 +126,44 @@ If your application runs in IIS, skip the rest of this section.
...
+
 Name                       | Value
 ---------------------------|------
 `CORECLR_ENABLE_PROFILING` | `1`
 `CORECLR_PROFILER`         | `{846F5F1C-F9AE-4B07-969E-05C26BC060D8}`
 
-For example, to set the environment variables from a batch file before starting your application:
+#### Windows Services
+
+To automatically instrument a Windows Service, set the environment variables for that Service in the Windows Registry. Create a multi-string value called `Environment` in the `HKLM\System\CurrentControlSet\Services\<Service Name>` key. Then set the key's data to the values in the table:
```
Comment (lucaspimentel) [inline content/en/tracing/setup/dotnet-core.md:138]: The convention is to use caps in the placeholders.

```suggestion
To automatically instrument a Windows Service, set the environment variables for that Service in the Windows Registry. Create a multi-string value called `Environment` in the `HKLM\System\CurrentControlSet\Services\<SERVICE NAME>` key. Then set the key's data to the values in the table:
```

File: content/en/tracing/setup/dotnet-core.md [hunk 12]
```diff
@@ -126,24 +126,44 @@ If your application runs in IIS, skip the rest of this section.
...
+Set the environment variables from a batch file before starting your application:
 
 ```bat
 rem Set environment variables
 SET CORECLR_ENABLE_PROFILING=1
 SET CORECLR_PROFILER={846F5F1C-F9AE-4B07-969E-05C26BC060D8}
 
 rem Start application
-dotnet.exe example.dll
+example.exe
```
Comment (lucaspimentel) [inline content/en/tracing/setup/dotnet-core.md:164]: Reverting this change.
```suggestion
dotnet.exe example.dll
```

File: content/en/tracing/setup/dotnet-framework.md [hunk 13]
```diff
@@ -64,26 +64,43 @@ If your application runs in IIS, skip the rest of this section.
...
+
 | Name                   | Value                                    |
 | ---------------------- | ---------------------------------------- |
 | `COR_ENABLE_PROFILING` | `1`                                      |
 | `COR_PROFILER`         | `{846F5F1C-F9AE-4B07-969E-05C26BC060D8}` |
 
-For example, to set the environment variables from a batch file before starting your application:
+#### Windows Services
+
+To automatically instrument a Windows Service, set the environment variables for that Service in the Windows Registry. Create a multi-string value called `Environment` in the `HKLM\System\CurrentControlSet\Services\<Service Name>` key. Then set the key's data to the values in the table:
```
Comment (lucaspimentel) [inline content/en/tracing/setup/dotnet-framework.md:76]: The convention is to use caps in the placeholders.

```suggestion
To automatically instrument a Windows Service, set the environment variables for that Service in the Windows Registry. Create a multi-string value called `Environment` in the `HKLM\System\CurrentControlSet\Services\<SERVICE NAME>` key. Then set the key's data to the values in the table:
```

File: content/en/tracing/setup/dotnet-framework.md [hunk 14]
```diff
@@ -64,26 +64,43 @@ If your application runs in IIS, skip the rest of this section.
...
+Set the environment variables from a batch file before starting your application:
 
 ```bat
 rem Set environment variables
 SET COR_ENABLE_PROFILING=1
 SET COR_PROFILER={846F5F1C-F9AE-4B07-969E-05C26BC060D8}
 
 rem Start application
-example.exe
+dotnet.exe example.dll
```
Comment (lucaspimentel) [inline content/en/tracing/setup/dotnet-framework.md:101]: Reverting this change. `dotnet.exe` is for .NET Core.
```suggestion
example.exe
```

File: content/en/tracing/compatibility_requirements/dotnet-core.md [hunk 15]
```diff
@@ -14,6 +14,8 @@ The .NET Tracer supports automatic instrumentation on .NET Core 2.1 and 3.1. It
 
 The .NET Tracer works on .NET Core 2.0, 2.2, and 3.0, but these versions reached their end of life and are no longer supported by Microsoft. See [Microsoft's support policy][3] for more details. We recommend using the latest patch version of .NET Core 2.1 or 3.1.  Older versions of .NET Core on Linux/x64 have JIT compiler bugs that can cause applications to throw exceptions when using automatic instrumentation. If your application is running on .NET Core 2.0, 2.1.0-2.1.11, or 2.2.0-2.2.5, we strongly recommend you update your .NET Core runtime. If you cannot update, you may need to set the environment variable `DD_CLR_DISABLE_OPTIMIZATIONS=true` to work around the issue. See [DataDog/dd-trace-dotnet/issues/302][4] for more details.
 
+**Note:** When using both manual and automatic instrumentation, it is important to keep the msi installer and NuGet package versions in sync.
```
Comment (sarina-dd) [inline content/en/tracing/compatibility_requirements/dotnet-core.md:17]:
```suggestion
**Note:** When using both manual and automatic instrumentation, it is important to keep the MSI installer and NuGet package versions in sync.
```

File: content/en/tracing/compatibility_requirements/dotnet-framework.md [hunk 16]
```diff
@@ -12,6 +12,8 @@ The .NET Datadog Trace library is open source - view the [Github repository][1]
 
 The .NET Tracer supports automatic instrumentation on .NET Framework 4.5 and above. It also supports [.NET Core][2].
 
+**Note:** When using both manual and automatic instrumentation, it is important to keep the msi installer and NuGet package versions in sync.
```
Comment (sarina-dd) [inline content/en/tracing/compatibility_requirements/dotnet-framework.md:15]:
```suggestion
**Note:** When using both manual and automatic instrumentation, it is important to keep the MSI installer and NuGet package versions in sync.
```

File: content/en/tracing/custom_instrumentation/dotnet.md [hunk 17]
```diff
@@ -17,9 +17,11 @@ further_reading:
       text: 'Explore your services, resources, and traces'
 ---
 <div class="alert alert-info">
-If you have not yet read the instructions for auto-instrumentation and setup, start with the <a href="https://docs.datadoghq.com/tracing/setup/dotnet/">.NET Setup Instructions</a>.
+For instructions on how to setup the .NET Tracer and enable automatic instrumentation, see the <a href="https://docs.datadoghq.com/tracing/setup/dotnet/">.NET Setup Instructions</a>.
```
Comment (sarina-dd) [inline content/en/tracing/custom_instrumentation/dotnet.md:20]:
```suggestion
For instructions on how to setup the .NET Tracer and enable automatic instrumentation, see the <a href="https://docs.datadoghq.com/tracing/setup/dotnet/">.NET setup instructions</a>.
```

File: content/en/tracing/custom_instrumentation/dotnet.md [hunk 18]
```diff
@@ -17,9 +17,11 @@ further_reading:
...
 </div>
 
+**Note:** When using both custom and automatic instrumentation, it is important to keep the msi installer and NuGet package versions in sync.
```
Comment (sarina-dd) [inline content/en/tracing/custom_instrumentation/dotnet.md:23]:
```suggestion
**Note:** When using both custom and automatic instrumentation, it is important to keep the MSI installer and NuGet package versions in sync.
```

File: content/en/tracing/setup/dotnet-core.md [hunk 19]
```diff
@@ -126,24 +126,44 @@ If your application runs in IIS, skip the rest of this section.
...
-For example, to set the environment variables from a batch file before starting your application:
+#### Windows Services
+
+To automatically instrument a Windows Service, set the environment variables for that Service in the Windows Registry. Create a multi-string value called `Environment` in the `HKLM\System\CurrentControlSet\Services\<Service Name>` key. Then set the key's data to the values in the table:
+```text
+CORECLR_ENABLE_PROFILING=1
+CORECLR_PROFILER={846F5F1C-F9AE-4B07-969E-05C26BC060D8}
+```
+
+
```
Comment (sarina-dd) [inline content/en/tracing/setup/dotnet-core.md:144]:
```suggestion
```

File: content/en/tracing/setup/dotnet-core.md [hunk 20]
```diff
@@ -203,6 +223,8 @@ For more details on manual instrumentation and custom tagging, see [Manual instr
 
 Manual instrumentation is supported on .NET Framework 4.5 and above on Windows and on .NET Core 2.1, 3.0, and 3.1 on Windows and Linux.
 
+**Note:** When using both manual and automatic instrumentation, it is important to keep the msi installer and NuGet package versions in sync.
```
Comment (sarina-dd) [inline content/en/tracing/setup/dotnet-core.md:226]:
```suggestion
**Note:** When using both manual and automatic instrumentation, it is important to keep the MSI installer and NuGet package versions in sync.
```

File: content/en/tracing/setup/dotnet-framework.md [hunk 21]
```diff
@@ -92,6 +109,8 @@ For more details on manual instrumentation and custom tagging, see [Manual instr
 
 Manual instrumentation is supported on .NET Framework 4.5 and above on Windows and on .NET Core 2.1, 3.0, and 3.1 on Windows and Linux.
 
+**Note:** When using both manual and automatic instrumentation, it is important to keep the msi installer and NuGet package versions in sync.
```
Comment (sarina-dd) [inline content/en/tracing/setup/dotnet-framework.md:112]:
```suggestion
**Note:** When using both manual and automatic instrumentation, it is important to keep the MSI installer and NuGet package versions in sync.
```
